src/error.rs

```diff
@@ -136,6 +136,8 @@ pub(crate) enum Error {
   Unstable { feature: &'static str },
   #[snafu(display("Torrent verification failed."))]
   Verify,
+  #[snafu(display("Failed to parse or write JSON: {}", source))]
+  Json { source: serde_json::Error },
```

Comment:
Can this be `JsonSerialize`, with the display text `Failed to serialize JSON: {}`?